src/StatusQ/Controls/StatusBaseButton.qml

```diff
@@ -19,158 +20,106 @@ Rectangle {
         Primary
     }
 
-    property StatusAssetSettings icon: StatusAssetSettings {
-        width: 24
-        height: 24
-    }
+    property StatusAssetSettings asset: StatusAssetSettings { }
 
     property bool loading: false
 
-    property alias hovered: sensor.containsMouse
+    property color normalColor
+    property color hoverColor
+    property color disabledColor
+
+    property color textColor
+    property color disabledTextColor
+    property color borderColor: "transparent"
 
     property int size: StatusBaseButton.Size.Large
     property int type: StatusBaseButton.Type.Normal
 
-    property alias text: label.text
-    property alias font: label.font
-
-    property real defaultLeftPadding: size === StatusBaseButton.Size.Large ? 24 : 12
-    property real defaultRightPadding: size === StatusBaseButton.Size.Large ? 24 : 12
-    property real defaultTopPadding: {
-        switch (size) {
-            case StatusBaseButton.Size.Tiny:
-              return 5
-            case StatusBaseButton.Size.Small:
-              return 10
-            case StatusBaseButton.Size.Large:
-            default:
-              return 11
-        }
-    }
-    property real defaultBottomPadding: {
-        switch (size) {
+    QtObject {
+        id: d
+        readonly property color textColor: root.enabled ? root.textColor : root.disabledTextColor
+        property int defaultHorizontalPadding: (((size === StatusBaseButton.Size.Large) ? 48 : 24) + root.horizontalPadding)
+        property int defaultVerticalPadding: {
+            var vertPadding;
+            switch (size) {
             case StatusBaseButton.Size.Tiny:
-              return 5
+                vertPadding =  10;
+                break;
             case StatusBaseButton.Size.Small:
-              return 10
+                vertPadding = 20;
+                break;
             case StatusBaseButton.Size.Large:
             default:
-              return 11
+                vertPadding = 22;
+                break;
+            }
+            return (vertPadding + root.verticalPadding);
         }
     }
 
+    implicitWidth: (layout.width + d.defaultHorizontalPadding)
+    implicitHeight: (layout.height + d.defaultVerticalPadding)
```

Comment:
Drop those... they are calculated automatically from the `contentItem` + `*padding`